internal/test/credentials/credentials.go

```diff
@@ -0,0 +1,80 @@
+package credentials
+
+import (
+	"fmt"
+	"golang.org/x/crypto/ssh"
+	"strings"
+)
+
+const (
+	// Username is the default windows username on AWS
+	Username = "Administrator"
+)
+
+// Credentials holds the information to access the Windows instance created.
+type Credentials struct {
+	// instanceID uniquely identifies the instanceID
+	instanceID string
+	// ipAddress contains the public ip address of the instance created
+	ipAddress string
+	// sshKey to access the instance created
+	sshKey ssh.Signer
+	// user used for accessing the  instance created
+	user string
+}
+
+// NewCredentials takes the instanceID, ip address and user of the Windows instance created and returns the
+// Credentials structure
+func NewCredentials(instanceID, ipAddress, user string) *Credentials {
+	return &Credentials{instanceID: instanceID, ipAddress: ipAddress, user: user}
+}
+
+// GetIPAddress returns the ip address of the given node
+func (cred *Credentials) GetIPAddress() string {
+	return cred.ipAddress
+}
+
+// GetSSHKey returns the SSH key associated with the given node
+func (cred *Credentials) GetSSHKey() ssh.Signer {
+	return cred.sshKey
+}
+
+// SetSSHKey sets the ssh signer for given node
+func (cred *Credentials) SetSSHKey(signer ssh.Signer) {
+	cred.sshKey = signer
+}
+
+// GetInstanceID returns the instanceId associated with the given node
+func (cred *Credentials) GetInstanceId() string {
+	return cred.instanceID
+}
+
+// GetUserName returns the user name associated with the given node
+func (cred *Credentials) GetUserName() string {
+	return cred.user
+}
+
+// String returns the string representation of Creds. This is required for Creds to be used with flags.
+func (c *Credentials) String() string {
+	return fmt.Sprintf("%v", *c)
+}
+
+// Set populates the list of credentials from the vmCreds command line argument
+func (c *Credentials) Set(value string) error {
+	if value == "" {
+		return nil
+	}
+
+	splitValue := strings.Split(value, ",")
+	// Credentials consists of three elements, so this has to be
+	if len(splitValue) != 2 {
+		return fmt.Errorf("incomplete VM credentials provided")
```

Comment:
`incorrect value for VM credentials` ?

Reply:
actually the condition checks if the credentials are in a pair of 2. So incomplete will better describe this error. incorrect might mean that the credentials provided are not correct (and not incomplete). This might cause confusion.

Reply:
I would then say that is a parsing error.